GWFish/modules/detection.py

```diff
@@ -141,6 +143,9 @@ def __init__(self, name: str, config=DEFAULT_CONFIG):
         fmax = eval(str(detector_def['fmax']))
         spacing = str(detector_def['spacing'])
 
+        self.L = eval(str(detector_def['arm_length']))
```

Comment:
It seems that the tests are failing because of this line here: it's evaluated for *any* detector, not just interferometers, so if you initialize e.g. LGWA it will fail.
It should be enough to move it under the block that runs only for interferometers!